### ISSUE TITLE
Use nginx to balance load between variation-normalization server processes

### ISSUE THREAD

**Issue by theferrit32** [clingen-data-model/architecture#564]
Right now we're using the default uvicorn method of distributing requests to worker processes, which is very "lightweight" in that it doesn't do any load-based routing, it just sends the requests evenly distributed to workers. So some workers may do a lot more work if their requests take longer.

We can use nginx and its `least_conn` upstream config to better balance load and help incoming requests get routed to idle workers. So instead of 1 uvicorn server with N workers, we will have N uvicorn servers with 1 worker, with each uvicorn bound to a different port, and nginx balancing the load between them.

https://www.uvicorn.org/deployment/#running-behind-nginx
https://docs.nginx.com/nginx/admin-guide/load-balancer/http-load-balancer/#choosing-a-load-balancing-method

**Comment by theferrit32** [comment]
Each QueryHandler object takes a significant amount of RAM. It looks like something like 400 MiB. So if each worker has a pool of them, this can get large quickly. I made the pool size configurable, so in our deployment I've reduced it to 1. Each uvicorn process still has like 460-480MiB of RAM usage right after it has started.

https://github.com/clingen-data-model/architecture/blob/242064f62a4ce0c95819ace87ae8ef9404d105b8/helm/charts/clingen-vicc/templates/variant-normalizer.yaml#L61
